### PR TITLE
fix(tracer): fix sampling on lambda

### PR DESCRIPTION
**Pull request by liadgo**

### CODE REVIEW

File: src/tracer.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ module.exports.createTracer = function createTracer() {
         trace: tracerObj,
         currRunner: null,
         pendingEvents: new Map(),
+        sampleRate: config.getConfig().sampleRate,
```
Comment (galbash) [inline src/tracer.js]: why set this on a per trace basis?

**Review by tvaintrob — DISMISSED** [review]

File: test/unit_tests/wrappers/test_openwhisk_traces.js [hunk 2]
```diff
@@ -46,9 +46,9 @@ describe('openwhiskWrapper tracer tests', () => {
             'addException'
         );
 
-        this.postTraceStub = sinon.stub(
+        this.sendTraceStub = sinon.stub(
```
Comment (galbash) [inline test/unit_tests/wrappers/test_openwhisk_traces.js]: why is this change required? did tests failed without it?

File: src/tracer.js [hunk 3]
```diff
@@ -364,6 +370,10 @@ function sendCurrentTrace(traceSender) {
 
     const sendResult = traceSender(traceJson);
     tracerObj.pendingEvents.clear();
+
+    if (tracerObj.sampleRate !== consts.DEFAULT_SAMPLE_RATE) {
+        tracerObj.deleted = true;
+    }
```
Comment (galbash) [inline src/tracer.js]: you can just check the config here instead of doing it on a per-trace basis (still set deleted on the current trace)

File: test/unit_tests/wrappers/test_openwhisk_traces.js (hunk 2 above)
Comment (liadgo) [reply test/unit_tests/wrappers/test_openwhisk_traces.js]: Yeah.
actually the entire test seems to be buggy.
TraceObj is not being initialised, therefore - no runner is being set (look at addRunner func), and the trace is will not be posted from now on  (which is a GOOD behavior) 
will fix that in the future.  @ranrib please approve.

File: src/tracer.js (hunk 3 above)
Comment (liadgo) [reply src/tracer.js]: Had this thought. for somereason chose to do this per-trace basis.
Looks ok with config. :)
changing!

**Review by galbash — APPROVED** [review]
